src/components/ComposerAttachments.vue

```diff
@@ -172,21 +284,44 @@ export default {
 
 			try {
 				const paths = await picker.pick(t('mail', 'Choose a file to add as attachment'))
-				const fileSizes = await Promise.all(paths.map(getFileSize))
-				const newTotal = sum(fileSizes) + this.totalSizeOfUpload()
+				// maybe fiiled front with placeholder loader...?
+				const filesFromCloud = await Promise.all(paths.map(getFileData))
+
+				const sum = filesFromCloud.reduce((sum, item) => {
+					return sum + item.size
+				}, 0)
+
+				const newTotal = sum + this.totalSizeOfUpload()
+				const th = this
 
 				if (this.uploadSizeLimit && newTotal > this.uploadSizeLimit) {
 					this.showAttachmentFileSizeWarning(paths.length)
 
 					return
 				}
 
-				this.emitNewAttachments(paths.map(function(name) {
-					return {
+				this.emitNewAttachments(paths.map(function(name, i) {
+					const _cloudFile = {
 						fileName: name,
 						displayName: trimStart('/', name),
 						type: 'cloud',
+						size: filesFromCloud[i].size,
+
 					}
+					const _toAttachmentData = {
+						finished: true,
+						imageBlobURL: th.generatePreview(_cloudFile),
+						total: filesFromCloud[i].size,
+						sizeString: th.formatBytes(filesFromCloud[i].size),
+						hasPreview: filesFromCloud[i]['has-preview'],
+						// dont know, may be it will be conflict if cloud & local has equal IDs?
+						id: filesFromCloud[i].fileid,
+						uploaded: 0,
+					}
+
+					th.attachments.push(Object.assign(_toAttachmentData, _cloudFile))
+
+					return _cloudFile
```

Comment:
```suggestion
				this.emitNewAttachments(paths.map((name, i) => {
					const _cloudFile = {
						fileName: name,
						displayName: trimStart('/', name),
						type: 'cloud',
						size: filesFromCloud[i].size,

					}
					const _toAttachmentData = {
						finished: true,
						imageBlobURL: this.generatePreview(_cloudFile),
						total: filesFromCloud[i].size,
						sizeString: this.formatBytes(filesFromCloud[i].size),
						hasPreview: filesFromCloud[i]['has-preview'],
						// dont know, may be it will be conflict if cloud & local has equal IDs?
						id: filesFromCloud[i].fileid,
						uploaded: 0,
					}

					this.attachments.push(Object.assign(_toAttachmentData, _cloudFile))

					return _cloudFile
```

This is captured inside an arrow closure.